src/store/datasets-layers.ts

```diff
@@ -0,0 +1,232 @@
+import { del, ref, set } from '@vue/composition-api';
+import vtkITKHelper from '@kitware/vtk.js/Common/DataModel/ITKHelper';
+import vtkImageData from '@kitware/vtk.js/Common/DataModel/ImageData';
+import vtkBoundingBox from '@kitware/vtk.js/Common/DataModel/BoundingBox';
+import { defineStore } from 'pinia';
+import { useImageStore } from '@/src/store/datasets-images';
+import { resample } from '../io/resample/resample';
+import { useDICOMStore } from './datasets-dicom';
+import {
+  DataSelection,
+  makeDICOMSelection,
+  makeImageSelection,
+  selectionEquals,
+  getImage,
+} from './datasets';
+import { useErrorMessage } from '../composables/useErrorMessage';
+import { Manifest, StateFile } from '../io/state-file/schema';
+
+// differ from Image/Volume IDs with a branded type
+type DataSelectionKey = string & { __type: 'DataSelectionKey' };
+export type LayerID = string & { __type: 'LayerID' };
+
+export type Layer = {
+  selection: DataSelection;
+  id: LayerID;
+};
+
+const toDataSelectionKey = (selection: DataSelection) => {
+  const id =
+    selection.type === 'dicom' ? selection.volumeKey : selection.dataID;
+  return `${selection.type}::${id}` as DataSelectionKey;
+};
+
+const toDataSelectionFromKey = (key: DataSelectionKey) => {
+  const [type, id] = key.split('::');
+  if (type === 'dicom') return makeDICOMSelection(id);
+  if (type === 'image') return makeImageSelection(id);
+
+  throw new Error('Unknown DataSelection key');
+};
+
+export const useLayersStore = defineStore('layer', () => {
+  type _This = ReturnType<typeof useLayersStore>;
+
+  const parentToLayers = ref<Record<DataSelectionKey, Layer[]>>({});
+  const layerImages = ref<Record<LayerID, vtkImageData>>({});
+
+  async function _addLayer(
+    this: _This,
+    parent: DataSelection | null | undefined,
+    source: DataSelection
+  ) {
+    if (!parent) {
+      console.warn('Tried to addLayer without parent data selection');
```

Comment:
What do you think about throwing an error? console.warn doesn't do much for us here compared to just throwing.

Reply:
Good idea.  Throwing now